File: jieun^,^/BOJ_13305번_주유소_Greedy.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ_13305번_주유소_Greedy {


    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        long[] dist = new long[N-1];
        long[] price = new long[N];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i=0; i<N-1; i++){
            dist[i] = Long.parseLong(st.nextToken());
        }

        st = new StringTokenizer(br.readLine());
        for(int i=0; i<N; i++){
            price[i] = Long.parseLong(st.nextToken());
        }

        long ans = 0;
        long minOil = price[0];

        for (int i=0; i<N-1; i++){
            if (minOil > price[i]){
                minOil = price[i];
            }
            ans += minOil * dist[i];
        }

        System.out.println(ans);
    }
}

File: jieun^,^/BOJ_13305번_주유소_Greedy.py
def solution():
    N = int(input())    # 도시의 개수

    dist = list(map(int, input().split()))  # 도시 간의 거리

    price = list(map(int, input().split())) # 각 도시에서 기름의 가격
    # [5, 2, 4, 1]

    ans = dist[0] * price[0]
    minOil = price[0]

    for n in range(1, N-1): # 1 ~ N-2
        if minOil>price[n]:
            minOil = price[n]
        ans += minOil * dist[n]

    print(ans)

    
solution()
